core/src/Streamly/Internal/Data/Serialize/TH.hs

```diff
@@ -362,69 +362,65 @@ deriveSerializeInternal preds headTy cons = do
             ]
     return [plainInstanceD preds (AppT (ConT ''Serialize) headTy) methods]
 
--- | Template haskell helper to create instances of 'Serialize' automatically.
+-- | Like 'deriveSerialize' but control which types variables should not get the
+-- 'Serialize' constraint and how to substitute the variables while applying to
+-- the main type.
```

Comment:
it is not obvious what substitution means here. Can it be explained better with a use case/example?

Reply:
Updated the doc.